util/src/main/java/io/grpc/util/AdvancedTlsX509KeyManager.java

```diff
@@ -39,20 +37,21 @@
 
 /**
  * AdvancedTlsX509KeyManager is an {@code X509ExtendedKeyManager} that allows users to configure
- * advanced TLS features, such as private key and certificate chain reloading, etc.
+ * advanced TLS features, such as private key and certificate chain reloading.
+ * We expect only one of {@link AdvancedTlsX509KeyManager#updateIdentityCredentials},
+ * {@link AdvancedTlsX509KeyManager#updateIdentityCredentialsFromFile
+ * (File, File, long, TimeUnit, ScheduledExecutorService)},
+ * {@link AdvancedTlsX509KeyManager#updateIdentityCredentialsFromFile
+ * (File, File, long, TimeUnit, ScheduledExecutorService)} methods to be called after instantiation
+ * of this class.
  */
-@ExperimentalApi("https://github.com/grpc/grpc-java/issues/8024")
 public final class AdvancedTlsX509KeyManager extends X509ExtendedKeyManager {
   private static final Logger log = Logger.getLogger(AdvancedTlsX509KeyManager.class.getName());
-
-  // The credential information sent to peers to prove our identity.
+  // Minimum allowed period for refreshing files with credential information.
+  public final static int MINIMUM_REFRESH_PERIOD = 1 ;
```

Comment:
Should this be marked `private`?

Reply:
I'd prefer to keep it public since I see few other constants are public as well, plus it's a part of implementation that we don't hide from the users.

Reply:
Can you point to these other examples?

From my point of view, we should not mark a field public unless there is a clear reason why a user needs to depend on this value. In this case, it's not clear to me what that reason is.

Reply:
The following 2 are public and I couldn't find any usages outside of tests. Maybe there is a reason why a user needs to depend on them but I'm not sure

https://github.com/grpc/grpc-java/blob/2bc4306940e7a2dea529be8a6ce536cab0638dc0/core/src/main/java/io/grpc/internal/GrpcUtil.java#L189

https://github.com/grpc/grpc-java/blob/2bc4306940e7a2dea529be8a6ce536cab0638dc0/alts/src/main/java/io/grpc/alts/internal/AltsTsiHandshaker.java#L39

Reply:
I think there are is a difference between those examples and this one, because those libraries are very much internal to gRPC, whereas this is a public API. Further, in the `AltsTsiHandshaker.java` case it really should be package-private and annotated with the "VisibleForTesting" annotation.

For a public API such as this one, I think we should not expose a constant like this one to the user unless we expect them to really need to have access to it. If there is such a reason in this case, please let me know. Otherwise, I do suggest we mark this private.

Reply:
fixed based on offline talk